dashboard/lti_new.py

```diff
@@ -0,0 +1,196 @@
+
+
+def extracting_launch_variables_for_tool_use(request, message_launch):
+    launch_id = message_launch.get_launch_id()
+    launch_data = message_launch.get_launch_data()
+    logger.debug(f'lti launch data {launch_data}')
+    custom_params = launch_data['https://purl.imsglobal.org/spec/lti/claim/custom']
+    logger.debug(f'lti_custom_param {custom_params}')
+    if not custom_params:
+        raise Exception(
+            f'You need have custom parameters configured on your LTI Launch. Please see the LTI installation guide on the Github Wiki for more information.'
+        )
+    course_name = launch_data['https://purl.imsglobal.org/spec/lti/claim/context']['title']
+    roles = launch_data['https://purl.imsglobal.org/spec/lti/claim/roles']
+    username = custom_params['user_username']
+    course_id = custom_params['canvas_course_id']
+    email = launch_data['email']
+    first_name = launch_data['given_name']
+    last_name = launch_data['family_name']
+    user_img = launch_data['picture']
+
+    # Add user to DB if not there; avoids Django redirection to login page
+    try:
+        user_obj = User.objects.get(username=username)
+    except User.DoesNotExist:
+        password = ''.join(random.sample(string.ascii_letters, settings.RANDOM_PASSWORD_DEFAULT_LENGTH))
+        user_obj = User.objects.create_user(username=username, email=email, password=password, first_name=first_name,
+                                            last_name=last_name)
+    user_obj.backend = 'django.contrib.auth.backends.ModelBackend'
+    django.contrib.auth.login(request, user_obj)
+    user_roles = course_user_roles(roles, username)
+    is_instructor = check_if_instructor(user_roles, username, course_id)
+    short_roles_list = short_user_role_list(user_roles)
+
+    course_details = None
+    is_course_data_loaded = False
+
+    try:
+        course_details = Course.objects.get(canvas_id=course_id)
+    except ObjectDoesNotExist:
+        if is_instructor or user_obj.is_staff:
+            canvas_long_id = canvas_id_to_incremented_id(course_id)
+            course_details = Course.objects.create(id=canvas_long_id, canvas_id=course_id, name=course_name)
+            CourseViewOption.objects.create(course_id=canvas_long_id)
+
+    if course_details is None:
+        logger.info(f'Course {course_id} data has not yet been loaded')
+    elif course_details.term_id is not None:
+        logger.info(f'Course {course_id} data is ready')
+        is_course_data_loaded = True
+
+    myla_globals = {
+        "username": username,
+        "is_superuser": user_obj.is_staff,
+        "login": settings.LOGIN_URL,
+        "logout": settings.LOGOUT_URL,
+        "user_image": user_img,
+        "primary_ui_color": settings.PRIMARY_UI_COLOR,
+        "help_url": settings.HELP_URL,
+        "google_analytics_id": settings.GA_ID,
+        "user_courses_info": [{"course_id": course_id, "course_name": course_name, "enrollment_type": short_roles_list}],
```

Comment:
`short_roles_list` is only used here, so remove it and use the function call here.

```suggestion
        'user_courses_info': [{
            'course_id': course_id,
            'course_name': course_name,
            'enrollment_type': short_user_role_list(user_roles)
        }],
```